src/plugins/rust/templates/tokenizer.template.rs

```diff
@@ -143,29 +151,42 @@ impl Tokenizer {
             .get(self.get_current_state())
             .unwrap();
 
-        for i in 0..lex_rules_for_state.len() {
-            let lex_rule = LEX_RULES[i];
+        let mut max_match_len = 0;
+        let mut max_match_token: Option<&'static str> = None;
+
+        for i in lex_rules_for_state {
+            let i = *i as usize;
 
-            if let Some(matched) = self._match(str_slice, &Regex::new(lex_rule).unwrap()) {
+            // the previous author use this to generate a new regex expression
+            // every time when `get_next_token` is called
+            // now this variable is of no use, I leave it here for debug use
+            let _lex_rule = LEX_RULES[i];
+            
+            if let Some(matched) = self._match(str_slice, &REGEX_RULES[i]) {
 
                 // Manual handling of EOF token (the end of string). Return it
                 // as `EOF` symbol.
                 if matched.len() == 0 {
                     self.cursor = self.cursor + 1;
                 }
 
-                self.yytext = matched;
-                self.yyleng = matched.len();
-
-                let token_type = self.handlers[i](self);
-
-                // "" - no token (skip)
-                if token_type.len() == 0 {
-                    return self.get_next_token();
+                // find longest match
+                if matched.len() > max_match_len {
+                    self.yytext = matched;
+                    self.yyleng = matched.len();
+                    max_match_len = matched.len();
+                    max_match_token = Some(self.handlers[i](self));
```

Comment:
This changes the semantics how other plugins work. IIRC, Bison can pass some option to differentiate the longest match vs. the non-greedy one. Can we move this part with the "longest match" to a separate PR?